### PR TITLE
test(compiler): Prevent back-sliding on already passing template pipeline tests

### PR DESCRIPTION
**Pull request by mmalerba**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
All tests run, even ones for features not yet implemented in the template pipeline, this prevented the tests from running on CI, as the status would be red.

Issue Number: N/A

## What is the new behavior?
Adds support to testing infrastructure to skip specific tests when running with the new template pipeline. This allows the already passing tests to run on CI.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by dylhunn — APPROVED** [review]
Nit: is the final commit intended to be empty?

**Comment by mmalerba** [comment]
Oh, I have a habit of using an empty first commit sometimes because on the components repo all of the commits get squashed under the first one when its merged. So I name the first commit according to our commit guidelines and then just name the rest whatever I feel like. I don't know if this repo works the same way with squashing, I can remove the empty one and rename the others if thats preferred

**Comment by dylhunn** [comment]
@mmalerba We don't do any such squashing on the fw repo; all commits will show up in the history

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "ng-dev": "ts-node --esm --project .ng-dev/tsconfig.json --transpile-only node_modules/@angular/ng-dev/bundles/cli.mjs",
     "build": "ts-node --esm --project scripts/tsconfig.json scripts/build/build-packages-dist.mts",
     "test": "bazelisk test",
-    "test:ci": "bazelisk test -- //... -//devtools/... -//aio/...",
+    "test:ci": "bazelisk test -- //... -//devtools/... -//aio/...; bazelisk test --//packages/compiler:use_template_pipeline //packages/compiler-cli/test/compliance/full",
```
Comment (devversion) [inline package.json:30]:
```suggestion
    "test:ci": "bazelisk test -- //... -//devtools/... -//aio/... && bazelisk test --//packages/compiler:use_template_pipeline //packages/compiler-cli/test/compliance/full",
```

**Review by devversion — APPROVED** [review]
LGTM one minor nit

**Comment by mmalerba** [comment]
Ok, I've manually squashed the commits under the first one

**Review by alxhub — APPROVED** [review]

**Comment by dylhunn** [comment]
@mmalerba I'm guessing you want `merge: caretaker note` if you're ready to merge this, since CI is misbehaving.

**Comment by pkozlowski-opensource** [comment]
Marking this PR as blocked since the CI didn't run on it.

**Comment by mmalerba** [comment]
I've resolved the CI issues 👍

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 9429f96f6294f5c0bbb29f8f153ff467754d244e.

**Pull request closed without merging**